src/messenger.rs

```diff
@@ -302,7 +302,9 @@ where
             request_api_key: R::API_KEY,
             request_api_version: body_api_version,
             correlation_id: Int32(correlation_id),
-            client_id: Some(NullableString(None)),
+            // Technically we don't need to send a client_id, but newer redpanda version fail to parse the message
+            // without it. See https://github.com/influxdata/rskafka/issues/169 .
+            client_id: Some(NullableString(Some(String::from(env!("CARGO_PKG_NAME"))))),
```

Comment:
Just to confirm, is this the fix for the redpanda problem (https://github.com/influxdata/rskafka/issues/169)? 

And the rest of this PR is unrelated to the compatibility problem, but helps eliminate the topic existence (or lack off) we see in testing?